lib/tasks/ci/hooks/pre-commit.py

```diff
@@ -4,62 +4,189 @@
+            raise BadGithubRepo
 
-    for line in content:
-        line = "".join(line.split())
-        for specifier in SPECIFIERS:
-            idx = line.find(specifier)
-            if idx < 0:
+        org = _repo[0]
+        repository = _repo[1]
+        gh_repo = self.api.repository(org, repository)
+
+        contents = gh_repo.directory_contents('/', return_as=dict)
+        files = {}
+        for entry, content in contents.iteritems():
+            if content.type != "dir":
+                files[content.name] = content
                 continue
 
-            req = line[:idx]
-            specifier = line[idx:]
-
-            if req in reqs and reqs[req][0] != specifier:
-                # version mismatch
-                print "There's a version mismatch with {req} " \
-                    " {spec} and {prev_spec} defined in {src}.".format(
-                        req=req,
-                        spec=specifier,
-                        prev_spec=reqs[req][0],
-                        src=reqs[req][1]
-                    )
-                sys.exit(ERR)
-            elif req not in reqs:
-                reqs[req] = (specifier, fname)
-                break
-
-
-requirements = {}
-files = get_files('requirements.txt')
-
-for f in files:
-    process_requirements(requirements, f)
-
-print "No requirement version conflicts found. Looking good... ;)"
-for req, spec in requirements.iteritems():
-    print "{req}{spec} first found in {fname}".format(
-        req=req,
-        spec=spec[0],
-        fname=spec[1]
-    )
-
-sys.exit(OK)
+            req = gh_repo.file_contents("/{}/requirements.txt".format(entry))
+            reqs[entry] = req.decoded
+
+        fmatches = []
+        for pattern in self.req_patterns:
+            fmatches.extend(fnmatch.filter(files.keys(), pattern))
+
+        for match in fmatches:
+            req = gh_repo.file_contents(files[match].path)
+            reqs[match] = req.decoded
+
+        return reqs
+
+    def get_local_files(self):
+        matches = []
+        for src in self.local_sources:
+            for pattern in self.req_patterns:
+                for root, dirnames, filenames in os.walk(src):
+                    for filename in fnmatch.filter(filenames, pattern):
+                        matches.append(os.path.join(root, filename))
+
+        return matches
+
+    def get_local_contents(self, files):
+        local_reqs = {}
+        for fname in files:
+            with open(fname) as f:
+                content = f.read()
+
+            local_reqs[fname] = content
+
+        return local_reqs
+
+    def get_all_requirements(self):
+        reqs = {}
+        if self.api:
+            for repo in self.remote_sources:
+                try:
+                    requirements = self.get_repo_requirements(repo)
+                    if requirements:
+                        reqs[repo] = requirements
+                except BadGithubRepo:
+                    pass
```

Comment:
Could you add a `print` warning? It's shouldn't be completely invisible, it made things hard to debug for me